### DISCUSSION TITLE
How to plot only prediction masks?

### DISCUSSION THREAD

**Discussion by sisaha9** [facebookresearch/detectron2#2570] [Q&A] [answered]
I am working based off this notebook: https://colab.research.google.com/github/Tony607/detectron2_instance_segmentation_demo/blob/master/Detectron2_custom_coco_data_segmentation.ipynb#scrollTo=U5LhISJqWXgM

I know the prediction masks are stored in the output["instances"].pred_masks. I am using the segmentation for obstacle avoidance. I want to use the segmentation boundaries (not the rectangular boxes) to find the largest gap between the boundaries so I can move my car there. But the pred_masks seems to be a 3 channel mask with Trues and Falses as opposed to the generally 1 channel masks which are grayscale that I am used to. Any help would be great

**Comment by sisaha9** [comment, marked as answer]
Oh wait I figured it out. Each instance can be accessed by using their index. Thanks!

**Reply by Nobuyuki-Enzan** [reply]
Could you please tell me how to work it out?
I have the same issue, and so far I don't figure it out.

**Reply by sisaha9** [reply]
I just summed up the pred_masks along the number of instances axis. That gives you a 2d mask which you can binarize to 0 and 255

**Reply by sisaha9** [reply]
To add on. Each channel in the mask represents the mask for a certain instance of one of your classes. So if you have 2 classes and 5 objects detected in an image you will have 5 channels. Using output["instances"].pred_masks[index] where index is 0-4 will give you the mask for each of the objects. If you say output["instances"].pred_masks.sum(axis = 0) it will sum along these channels to give you 1 prediction mask with all the objects in it. The values will range from 0 - number of channels. I just binarized it to wherever it is > 0, make it 255 otherwise leave it as 0. I am now able to use this for the guidance calculations I was doing with previous basic OpenCV masking

**Reply by Nobuyuki-Enzan** [reply]
Thank you four your prompt reply. I'll try it.

**Reply by lzh1998-jansen** [reply]
hello，i want to know  how to calculate the area of mask after i got predicted?
